sdk/identity/azure-identity/tests/test_default.py

```diff
@@ -99,6 +99,8 @@ def send(request, **_):
     # all credentials not representing managed identities should use a specified authority or default to public cloud
     exercise_credentials("authority.com")
     exercise_credentials(None, KnownAuthorities.AZURE_PUBLIC_CLOUD)
+    with patch('os.environ', {EnvironmentVariables.AZURE_AUTHORITY_HOST: "login.microsoftonline.com"}):
```

Comment:
`"login.microsoftonline.com" == KnownAuthorities.AZURE_PUBLIC_CLOUD`, so you're setting the environment variable to the default value when the environment variable has no value i.e. this test would pass if the credential ignored the environment variable. I suggest `localhost` as a better test value.